sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -67,24 +67,30 @@ def csr_mean_variance_axis0(X):
         Feature-wise variances
 
     """
-    cdef unsigned int n_samples = X.shape[0]
-    cdef unsigned int n_features = X.shape[1]
+    return _csr_mean_variance_axis0(X.data, X.shape, X.indices)
 
-    cdef np.ndarray[DOUBLE, ndim=1, mode="c"] X_data
-    X_data = np.asarray(X.data, dtype=np.float64)     # might copy!
-    cdef np.ndarray[int, ndim=1] X_indices = X.indices
+
+def _csr_mean_variance_axis0(np.ndarray[floating, ndim=1, mode="c"] X_data,
+                             shape,
+                             np.ndarray[int, ndim=1] X_indices):
```

Comment:
You could write a small comment if we want to figure out why you wrote this wrapper. (That is, you can pass arguments that could have different dtypes only as arguments to functions and in this case floating identifies the dtype of the argument passed. The same cannot be done with variables)